src/test/tuntap.cpp

```diff
@@ -152,7 +155,128 @@ TEST(tuntap, read_from_tun) {
 		tuntap.read_from_tun(receive_buffer.data(), receive_buffer.size())
 		,0U
 	);
+}
+
+TEST(tuntap, read_from_tun_separated_addresses) {
+	c_tuntap_linux_obj tuntap;
+	auto normal_process =
+	[](const std::array<boost::asio::mutable_buffer, 4> &bufs) -> size_t {
+		std::array<unsigned char, IPV6_LEN> src, dst;
+		std::array<unsigned char, 71> packet;
+		std::tie(src, dst, packet) = get_separated_packet();
+		// check buffers siazes
+		EXPECT_EQ(boost::asio::buffer_size(bufs.at(0)), 8U);
+		EXPECT_EQ(boost::asio::buffer_size(bufs.at(1)), IPV6_LEN);
+		EXPECT_EQ(boost::asio::buffer_size(bufs.at(2)), IPV6_LEN);
+		EXPECT_EQ(boost::asio::buffer_size(bufs.at(3)), 71U - 8);
```

Comment:
add comment:

size:
8 B beginning of header (flow etc)
16 B ipb6 addr 
16 B ipb6 addr
71 is merit (packet - 2*16)
and the rest is merit - beginning, so 71-8